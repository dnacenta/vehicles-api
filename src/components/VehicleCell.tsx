import {Text, View, Image, StyleSheet} from 'react-native';
import {RectButton} from 'react-native-gesture-handler';

type VehicleCellProps = {
  onPress(): void;
  image: string;
  name: string;
};

const VehicleCell = ({onPress, image, name}: VehicleCellProps) => {
  return (
    <RectButton onPress={() => onPress()}>
      <View style={styles.cellContainer}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            source={{uri: image}}
          />
        </View>
        <View style={styles.textContainer}>
          <Text style={styles.text}>{name}</Text>
        </View>
      </View>
    </RectButton>
  );
};

const styles = StyleSheet.create({
  cellContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    margin: 8,
    backgroundColor: '#FDFE9E',
    padding: 8,
    borderRadius: 4,
    borderColor: '#14544E',
    borderWidth: 1,
  },
  imageContainer: {
    width: 80,
    height: 80,
    borderRadius: 4,
    backgroundColor: '#ffffff',
  },
  image: {
    width: 80,
    height: 80,
    borderRadius: 4,
    borderColor: '#14544E',
    borderWidth: 1,
  },
  textContainer: {
    flex: 3 / 4,
    marginLeft: 16,
  },
  text: {
    fontSize: 18,
    fontWeight: '600',
  },
});

export default VehicleCell;
