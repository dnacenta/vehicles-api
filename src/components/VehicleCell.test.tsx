import { render, screen } from "@testing-library/react-native";
import VehicleCell from "./VehicleCell";

describe("<VehicleCell />", () => {
  it("matches snapshot", () => {
    render(
      <VehicleCell
        onPress={jest.fn()}
        image="http://loremflickr.com/640/640/transport?70161"
        name="lorem"
      />
    );

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
