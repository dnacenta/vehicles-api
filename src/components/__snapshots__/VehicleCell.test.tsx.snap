// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehicleCell /> matches snapshot 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.105}
  collapsable={false}
  delayLongPress={600}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[Function]}
  style={{}}
  underlayColor="black"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "black",
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderRadius": undefined,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 4,
        "flex": 1,
        "flexDirection": "row",
        "margin": 8,
        "padding": 8,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#ced6e0",
          "borderRadius": 4,
          "height": 80,
          "width": 80,
        }
      }
    >
      <Image
        source={
          {
            "uri": "http://loremflickr.com/640/640/transport?70161",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 80,
            "width": 80,
          }
        }
      />
    </View>
    <View
      style={
        {
          "flex": 0.75,
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        lorem
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
